pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -643,7 +647,41 @@ func podAffinity(i *infinispanv1.Infinispan, matchLabels map[string]string) *cor
 	return i.Spec.Affinity
 }
 
-// deploymentForInfinispan returns an infinispan Deployment object
+func (r *ReconcileInfinispan) updatePodsLabels(m *infinispanv1.Infinispan, podList *corev1.PodList) error {
+	if len(podList.Items) == 0 {
+		return nil
+	}
+
+	labelsForPod := PodLabels(m.Name)
+	m.AddOperatorLabelsForPods(labelsForPod)
+	m.AddLabelsForPods(labelsForPod)
+
+	for _, pod := range podList.Items {
```

Comment:
Why are we updating pod's individually instead of just updating the StatefulSet spec?

Reply:
Updating `Spec.PodTemplateSpec` will cause a rolling update of the whole cluster just for Pods updating that's necessary in our case

Reply:
As the update is only to the metadata labels, does this not cause the individual pods to be restarted?

Reply:
Which update? Any StatefulSet `PodTemplateSpec` update including annotations or labels will cause rolling update of whole cluster. For individual Pods update only annotations or labels will be updated.

Reply:
Sorry I meant "As the update is only to the metadata labels of the *individual pods*, does this not cause the individual pods to be restarted?". But I guess not :slightly_smiling_face: